crates/router/src/connector/utils.rs

```diff
@@ -541,6 +541,25 @@ impl<Flow, Request, Response> RouterData for types::RouterData<Flow, Request, Re
     }
 }
 
+pub trait GetAddressDetails {
+    fn get_email(&self) -> Result<Email, Error>;
+    fn get_phone(&self) -> Result<Secret<String>, Error>;
+}
+
+impl GetAddressDetails for api::Address {
+    fn get_email(&self) -> Result<Email, Error> {
+        self.email.clone().ok_or_else(missing_field_err("email"))
+    }
+
+    fn get_phone(&self) -> Result<Secret<String>, Error> {
+        self.phone
+            .clone()
+            .map(|phone_details| phone_details.get_number_with_country_code())
+            .transpose()?
+            .ok_or_else(missing_field_err("phone"))
+    }
```

Comment:
This function should only return the phone details, Adding a country-code in prefix is a different operation, should not be a part of this function.